chiselFrontend/src/main/scala/chisel3/core/Mem.scala

```diff
@@ -143,7 +143,7 @@ object SeqMem {
   * @note when multiple conflicting writes are performed on a Mem element, the
   * result is undefined (unlike Vec, where the last assignment wins)
   */
-sealed class SeqMem[T <: Data](t: T, n: Int) extends MemBase[T](t, n) {
+sealed class SyncReadMem[T <: Data](t: T, n: Int) extends MemBase[T](t, n) {
   def read(addr: UInt, enable: Bool): T = {
```

Comment:
Is this not a backwards incompatible change in the compatibility layer as well? What if someone matches on the type of a Mem? Obviously they will get a compiler error but yeah.